### ISSUE TITLE
Configurable filter logic

### ISSUE THREAD

**Issue by Etroid** [AddSearch/search-ui#78]
When using checkbox filters, I've noticed that the logic defaults to "OR". Is there a way to change this to "AND" for specific use cases?

```
searchui.filters({
        containerId: 'checkbox-container',
        type: AddSearchUI.FILTER_TYPE.CHECKBOX_GROUP,
        options: {
          pricing: {
            label: 'Pricing',
            filter: {"category": "1xpricing"}
          },
          partners  : {
            label: 'Partners',
            filter: {"category": "1xpartners"}
          },
          customers: {
            label: 'Customers',
            filter: {"category": "1xcustomers"}
          }
        }
      });
```

In the above example I'd like each selection to filter the result list down. If I select pricing and partners I want to see content tagged with both.

**Comment by anttiai** [comment]
Hey. First of all, you should use [custom fields](https://www.addsearch.com/docs/indexing/defining-custom-fields/) for filtering. In your example, simple URL filters (like 1xpricing) mean that the first part (i.e. _1x_) of the URL path equals _pricing_ (e.g. domain.com/pricing). As documents can't have multiple URLs, the AND filter (1xpricing AND 1xpartners) wouldn't match anything.

Anyway, there are two ways to achieve what you asked:

1) The easiest way is to create a separate filter component for each filter. Filters in a single filter component share OR logic, but the default logic across different components is AND.

```js
// Pricing
searchui.filters({
  containerId: 'checkbox-container-pricing',
  type: AddSearchUI.FILTER_TYPE.CHECKBOX_GROUP,
  options: {
    pricing: {
      label: 'Pricing',
      filter: {"category": "1xpricing"}
    }
  }
});

// Partners
searchui.filters({
  containerId: 'checkbox-container-partners',
  type: AddSearchUI.FILTER_TYPE.CHECKBOX_GROUP,
  options: {
    pricing: {
      label: 'Partners',
      filter: {"category": "1xpartners"}
    }
  }
});

// etc..
```

2) It's also possible to override the function that constructs the filtering object in the SearchUI library. The default [createFilterObject](https://github.com/AddSearch/search-ui/blob/master/src/components/filters/filterstateobserver.js#L12) can be copied, modified, and passed to AddSearchUI constructor in a setting named _createFilterObjectFunction_:

```js
var conf = {
  createFilterObjectFunction: customCreateFilterObject
};

var searchui = new AddSearchUI(client, conf);
```

Basically, you would need to modify the OR operator to AND [here](https://github.com/AddSearch/search-ui/blob/master/src/components/filters/filterstateobserver.js#L27).

By the way, Facets could be a valid option to achieve what you asked as well.
